### PR TITLE
Feature/trace all

### PR DESCRIPTION
**Pull request by karlthepagan**
this adds stacktraces to ... EVERYTHING

possibly useful for new users to understand what's being printed

might be able to add this conditionally to trace

### CODE REVIEW

File: util.js [hunk 1]
```diff
@@ -221,7 +232,7 @@ module.exports = {
             }
         }
         
-        console.log(Game.time, Util.dye(CRAYON.error, category), ...msg, Util.dye(CRAYON.birth, JSON.stringify(entityWhere)));
+        console.log(Game.time, Util.dye(CRAYON.error, category), ...msg, Util.dye(CRAYON.birth, JSON.stringify(entityWhere)), Util.stack());
```
Comment (Spedwards) [inline util.js]: Would this not stop behaviour when `DEBUG_STACKS` is true? Isn't trace also used for debugging, not just errors?

Comment (karlthepagan) [inline util.js]: It won't change when things are printed. My intent was to add this stacktrace to every print in order to find where some odd messages were coming from.

File: util.js [hunk 2]
```diff
@@ -176,9 +176,20 @@ module.exports = {
...
+
+    /**
+     * Build a stacktrace if DEBUG_STACKS or the first argument is true.
+     */
+    stack(force = false) {
+        if (DEBUG_STACKS || force) {
+            return new Error(`STACK; param:${DEBUG_STACKS}, force:${force}`).stack;
+        }
+
+        return ' ';
```
Comment (chobobobo) [inline util.js]: this appears to be appending a single space to every log message now?

Comment (karlthepagan) [inline util.js]: This was the least bad option  :(

Undefined and empty string both show up in console.log

Comment (Spedwards) [inline util.js]: What's wrong with empty string showing up?

Comment (chobobobo) [inline util.js]: it offends my delicate sensibilities.

Comment (karlthepagan) [inline util.js]: It looks like:
'''
Something something log normal here. Ok! ""
'''

Comment (Spedwards) [inline util.js]: `String.fromCodePoint(0x2060)`. It's a type of space that literally displays nothing.

> The word joiner does not produce any space, and prohibits a line break at its position.

https://en.wikipedia.org/wiki/Word_joiner

Comment (karlthepagan) [inline util.js]: I like the way you think buuuuuuuuut

I considered it but the console log also goes into a file and people might want to parse it.

File: util.js [hunk 3]
```diff
@@ -176,9 +176,20 @@ module.exports = {
...
         const text = Util.dye(CRAYON.system, roomName);
-        console.log(Util.dye(CRAYON.system, `<a href="/a/#!/room/${roomName}">${text}</a> &gt;`), ...message);
+        console.log(Util.dye(CRAYON.system, `<a href="/a/#!/room/${roomName}">${text}</a> &gt;`), ...message, Util.stack());
     },
-    
+
+    /**
+     * Build a stacktrace if DEBUG_STACKS or the first argument is true.
+     */
+    stack(force = false, placeholder = ' ') {
```
Comment (karlthepagan) [inline util.js]: just for you @Spedwards  :)

Comment (chobobobo) [inline util.js]: You guys make me laugh

**Comment by chobobobo** [comment]
Is this the intended outcome?
```
Error: STACK; param:true, force:false
    at Object.stack (util:187:20)
    at Object.mod.logSystem (global:293:106)
    at wrapper (/opt/engine/node_modules/lodash/index.js:3095:19)
    at Object.totalCPU (global:525:13)
    at Object.module.exports.loop (main:340:7)
    at __module (__mainLoop:1:52)
    at __mainLoop:2:3
    at ContextifyScript.Script.runInContext (vm.js:35:29)
    at Object.exports.evalCode (/opt/engine/dist/core/index.js:1462:76)
    at Object.exports.runCode (/opt/engine/dist/game/game.js:479:24)
```

**Comment by karlthepagan** [comment]
I'll move it to the next line and add a warning to the param.

**Comment by karlthepagan** [comment]
Updated to solve @ryagas 's issue

File: util.js [hunk 4]
```diff
@@ -176,9 +176,20 @@ module.exports = {
...
+        if (DEBUG_STACKS || force) {
+            return new Error(`STACK; param:${DEBUG_STACKS}, force:${force}`).stack;
```
Comment (chobobobo) [inline util.js]: I believe this is where the newline could be added so it doesn't appear inline.

**Comment by chobobobo** [comment]
@karlthepagan this PR is ready to merge with a very minor edit to output

**Pull request merged**